packages/react-components/src/components/VideoGallery/VideoGallery.tsx

```diff
@@ -189,106 +233,67 @@ export const VideoGallery = (props: VideoGalleryProps): JSX.Element => {
-      onRenderAvatar,
-      showMuteIndicator
-    } = props;
-
-    useEffect(() => {
-      if (isAvailable && !renderElement) {
-        onCreateRemoteStreamView && onCreateRemoteStreamView(userId, remoteVideoViewOption);
-      }
-      if (!isAvailable) {
-        onDisposeRemoteStreamView && onDisposeRemoteStreamView(userId);
-      }
-    }, [
-      isAvailable,
-      onCreateRemoteStreamView,
-      onDisposeRemoteStreamView,
-      remoteVideoViewOption,
-      renderElement,
-      userId
-    ]);
-
-    useEffect(() => {
-      return () => {
-        onDisposeRemoteStreamView && onDisposeRemoteStreamView(userId);
-      };
-    }, [onDisposeRemoteStreamView, userId]);
-
-    const renderVideoStreamElement = useMemo(() => {
-      // Checking if renderElement is well defined or not as calling SDK has a number of video streams limitation which
-      // implies that, after their threshold, all streams have no child (blank video)
-      if (!renderElement || !renderElement.childElementCount) {
-        // Returning `undefined` results in the placeholder with avatar being shown
-        return undefined;
-      }
-
-      return <StreamMedia videoStreamElement={renderElement} />;
-      // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [renderElement, renderElement?.childElementCount]);
-
+  } else {
     return (
-      <Stack className={gridStyle} key={userId} grow>
-        <VideoTile
-          userId={userId}
-          renderElement={renderVideoStreamElement}
-          displayName={displayName}
-          onRenderPlaceholder={onRenderAvatar}
-          isMuted={isMuted}
-          isSpeaking={isSpeaking}
-          showMuteIndicator={showMuteIndicator}
-        />
-      </Stack>
+      <div ref={containerRef}>
+        <Stack grow styles={videoGalleryContainerStyle}>
+          <GridLayout styles={styles ?? emptyStyles}>
+            <Stack
+              data-ui-id={ids.videoGallery}
+              horizontalAlign="center"
+              verticalAlign="center"
+              className={gridStyle}
+              grow
+            >
+              {localParticipant && defaultOnRenderLocalVideoTile}
+            </Stack>
+            {defaultOnRenderRemoteParticipants}
+          </GridLayout>
+          {audioParticipants && (
+            <Stack style={{ minHeight: isMobileScreen ? '6rem' : '8rem', maxHeight: isMobileScreen ? '6rem' : '8rem' }}>
+              <HorizontalGallery
+                onCreateRemoteStreamView={onCreateRemoteStreamView}
+                onDisposeRemoteStreamView={onDisposeRemoteStreamView}
+                onRenderAvatar={onRenderAvatar}
+                onRenderRemoteVideoTile={onRenderRemoteVideoTile}
+                participants={audioParticipants}
+                remoteVideoViewOption={remoteVideoViewOption}
+                showMuteIndicator={showMuteIndicator}
+              />
+            </Stack>
+          )}
+        </Stack>
+      </div>
     );
   }
-);
+};
```

Comment:
This function is over 200 lines long, can we reduce it?

Reply:
This will be reduced when we merge @mgamis-msft and my layouts...

Reply:
So looks like we won't be able to shrink the code further even after using the new layouts. 
Do you have any recommendations?
I don't see a way to move out the code as all of it is video gallery specific UI logic.

Reply:
Large chunks of components should be split out into isolated components, e.g.
```
return shouldFloatLocalVideo()) ?  <VideoGalleryWithFloatingLocalVideo />
 :  <VideoGalleryNAnotherName/>;

```